### PR TITLE
Fix issue 20599 - cpp_long as enum type doesn't work

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Comment by thewilsonator** [comment]
cc @TurkeyMan

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -2365,7 +2365,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             if (auto te = ed.memtype.isTypeEnum())
             {
                 EnumDeclaration sym = cast(EnumDeclaration)te.toDsymbol(sc);
-                if (!sym.memtype || !sym.members || !sym.symtab || sym._scope)
+                if (!sym.memtype || (!sym.members && !sym.isSpecial()) || !sym.symtab || sym._scope)
```
Comment (maxhaton) [inline src/dmd/dsymbolsem.d:2368]: This line is hard to understand at a glance, a comment would be nice (i.e. It isn't immediately obvious why this fixes the bug)

Comment (thewilsonator) [reply src/dmd/dsymbolsem.d]: done

File: src/dmd/dsymbolsem.d [hunk 2]
```diff
@@ -2365,7 +2365,10 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             if (auto te = ed.memtype.isTypeEnum())
             {
                 EnumDeclaration sym = cast(EnumDeclaration)te.toDsymbol(sc);
-                if (!sym.memtype || !sym.members || !sym.symtab || sym._scope)
+                // Special enums like __c_[u]long[long] are fine to forward reference
+                // see https://issues.dlang.org/show_bug.cgi?id=20599
+                if (sym.isSpecial()) {}
+                else if (!sym.memtype || (!sym.members) || !sym.symtab || sym._scope)
```
Comment (ibuclaw) [inline src/dmd/dsymbolsem.d:2371]: Why not just bundle it all into one condition `if (!sym.isSpecial() && (!sym.memtype || ...))`?

Comment (thewilsonator) [reply src/dmd/dsymbolsem.d]: done

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**